cmd/thanos/store.go

```diff
@@ -225,12 +230,14 @@ func runStore(
 		return errors.Wrap(err, "create index cache")
 	}
 
+	ignoreDeletionMarkFilter := block.NewIgnoreDeletionMarkFilter(logger, bkt, deleteDelay.Seconds()/2)
 	prometheusRegisterer := extprom.WrapRegistererWithPrefix("thanos_", reg)
 	metaFetcher, err := block.NewMetaFetcher(logger, fetcherConcurrency, bkt, dataDir, prometheusRegisterer,
 		block.NewTimePartitionMetaFilter(filterConf.MinTime, filterConf.MaxTime).Filter,
 		block.NewLabelShardedMetaFilter(relabelConfig).Filter,
 		block.NewConsistencyDelayMetaFilter(logger, consistencyDelay, prometheusRegisterer).Filter,
-		block.NewDeduplicateFilter().Filter,
+		ignoreDeletionMarkFilter.Filter,
+		block.NewDeduplicateFilter(ignoreDeletionMarkFilter).Filter,
```

Comment:
I don't think `NewDeduplicateFilter` needs `ignoreDeletionMarkFilter`.

Reply:
The rationale of why `NewDeduplicateFilter` required `ignoreDeletionMarkFilter`:
`syncer.GarbageCollect` calls `DeduplicateFilter.duplicateIds` and then marks them for deletion.

Now here we can ignore the blocks which are already marked for deletion, so that is the reason why `ignoreDeletionMarkFilter` is passed to `NewDeduplicateFilter`, because it needs to be used in DuplicateIds function.

If you want me to pass `ignoreDeletionMarkFilter` in `DuplicateIDs` function, then I am happy to do so as well. That would also be good.

Reply:
I am now passing `ignoreDeletionMarkFilter` in `DuplicateIDs` function since that is more clear according to me. Still not resolving this thread in case you have some further points to add or some other idea in mind.

Reply:
Since the latest feedback doesn't have any changes requested for this, I am resolving this. Please create a new thread if things need to be taken care of.